incubating/jira-issue-manager/script/jira_issue_manager.py

```diff
@@ -0,0 +1,310 @@
+
+    issue_summary = StepUtility.getEnvironmentVariable('ISSUE_SUMMARY', env)
+    issue_description = StepUtility.getEnvironmentVariable('ISSUE_DESCRIPTION', env)
+
+    # Retrieve the type environment variable and add the type to a dict representation
+    issue_type_env = StepUtility.getEnvironmentVariable('ISSUE_TYPE', env)
+    issue_type = None
+    if issue_type_env:
+        issue_type = "{'name': '" + issue_type_env + "'}"
+
+    # Retrieve the components, split the list, and create an array of dicts with each component
+    issue_components_env = StepUtility.getEnvironmentVariable('ISSUE_COMPONENTS', env)
+    issue_components = None
+    if issue_components_env:
+        issue_components = []
+        split_components = issue_components_env.split(",")
+        for component in split_components:
+            component_string = "{'name': '" + component + "'}"
+            issue_components.append(ast.literal_eval(component_string))
+
+    # Retrieve the comment information
+    existing_comment_id = StepUtility.getEnvironmentVariable('JIRA_COMMENT_ID', env)
+    comment_body = StepUtility.getEnvironmentVariable('COMMENT_BODY', env)
+
+    # Desired Jira status information
+    status = StepUtility.getEnvironmentVariable('DESIRED_ISSUE_STATUS', env)
+    jql_query = StepUtility.getEnvironmentVariable('JQL_QUERY', env)
+    jql_query_max_results = 50
+    jql_query_max_results_env = StepUtility.getEnvironmentVariable('JQL_QUERY_MAX_RESULTS', env)
+    if jql_query_max_results_env:
+        jql_query_max_results = jql_query_max_results_env
+
+    verbose = StepUtility.getEnvironmentVariable('VERBOSE', env)
+
+    current_environment = Environment(
+        jira_base_url,
+        jira_username,
+        jira_api_key,
+        action,
+        issue,
+        issue_project,
+        issue_summary,
+        issue_description,
+        issue_type,
+        issue_components,
+        existing_comment_id,
+        comment_body,
+        status,
+        jql_query,
+        jql_query_max_results,
+        verbose)    
+    return current_environment
+
+
+def authentication(current_environment):
+    # Basic authentication with an API Token
+    jira = JIRA(
+        current_environment.jira_base_url, 
+        basic_auth=(current_environment.jira_username, current_environment.jira_api_key)
+    )
+    return jira
+
+
+def step_action(action, authenticated_jira, current_environment):
+    print("Step Action: " + action)
+    actions = {
+        'issue_create': create_issue,
+        'issue_update': update_issue,
+        'issue_transition': transition_issue,
+        'issue_transition_and_update': transition_issue,
+        'comment_create': create_comment,
+        'comment_update': update_comment,
+        'verify_status': verify_issue_status,
+        'update_all_from_jql_query': update_multiple_issues,     
+    }
+    actions.get(action, action_required)(authenticated_jira, current_environment)
```

Comment:
Hey Brandon! You might want to split up the dictionary .get() and calling the resulting function reference.  If the user typos the action, then the get() will return a None, which will throw a TypeError when it gets called as a function. 
```suggestion
  action_func = actions.get(action, action_required)
  if action_func:
      action_func(authenticated_jira, current_environment)
  else:
      StepUtility.printFail("Exiting Step - invalid ACTION environment variable")
      sys.exit(1)
```